app/views/assessments/_edit_problems.html.erb

```diff
@@ -1,7 +1,35 @@
-<script type="text/javascript">
-  is_positive_grading = <%= @is_positive_grading %>;
-  has_annotations = <%= @has_annotations %>;
-</script>
+<% content_for :javascripts do %>
+    <script type="text/javascript">
+        is_positive_grading = <%= @is_positive_grading %>;
+        has_annotations = <%= @has_annotations %>;
+
+        document.addEventListener('DOMContentLoaded', function() {
+            var starToggleIcons = document.querySelectorAll('.star-icon');
+
+            starToggleIcons.forEach(function(icon) {
+                icon.addEventListener('click', function() {
+                    toggleStar(icon);
+                });
+            });
+        });
+
+        function toggleStar(icon) {
+            var starred = icon.getAttribute('data-starred');
+            starred = (starred === 'yes') ? 'no' : 'yes';
+            icon.innerHTML = (starred === 'yes') ? 'star' : 'star_border';
+            icon.setAttribute('data-starred', starred);
+            var starred_bool = (starred==='yes') ? 'true' : 'false';
+            // Update the corresponding problem.starred attribute
+            var problemId = icon.getAttribute('data-problem-id');
+            var dataUrl = icon.parentNode.parentNode.getAttribute('data-url');
+            $.ajax({
```

Comment:
There is some subtlety with the use of `ajax` here due to what the `update` method in `problems_controller` does.
1. it sets a `flash`. Usually, when a user saves a problem, they are redirected to the list of problems and so this would appear for the user. However, when using Ajax, the flash is set but not immediately "consumed", so it only shows up on the next page the user loads.
2. It performs a `redirect_to`. Usually, this is what redirects the user back to the list of problems. But when we are performing an Ajax call, it seems to follow the redirect also with the `patch` method, which results in an error.

![Screenshot 2024-03-12 at 13 35 20](https://github.com/autolab/Autolab/assets/9074856/3dccdbb4-a908-42e6-ac2b-73443a735ec9)

<img width="486" alt="Screenshot 2024-03-12 at 13 35 34" src="https://github.com/autolab/Autolab/assets/9074856/6fd467ed-194e-41e4-aac5-494d726a1c3b">

Of course, the starred flag is still updated, but this isn't ideal.

------------------

If we do not want to add a new `update` method just for `Ajax` calls, then perhaps we should use params to distinguish between Ajax updates and "normal" updates.

On the frontend side, this can be specified by adding the params to the end of a path helper, e.g. `course_assessment_problem_path(@course, @assessment, problem, ajax: true)`

On the backend side, we can check if the flag is set with something like `params.include?(:ajax)`. Then you can avoid setting flashes and redirecting in this scenario.